internal/db/schema/migrations/postgres/14/01_event.up.sql

```diff
@@ -0,0 +1,348 @@
+├─────────────────┤     │        │           │    
+│ public_id       │┼────┘        ○           │    
+│ sink_id         │             ╱│╲          │    
+│ event_type      │     ┌─────────────────┐  │    
+│ format_type     │     │event_stderr_sink│  │    
+│ allow_filters   │     ├─────────────────┤  │    
+│ deny_filters    │     │ public_id       │  │    
+│ path            │     │ sink_id         │  │    
+│ file_name       │     │ event_type      │┼─┘    
+│ rotate_bytes    │     │ format_type     │       
+│ rotate_duration │     │ allow_filters   │       
+│ rotate_max_files│     │ deny_filters    │       
+└─────────────────┘     └─────────────────┘       
+         ┼                        ┼               
+         │                        │               
+         │                        │               
+         │  ┌─────────────────┐   │               
+         │ ╱│event_format_type│╲  │               
+         └──├─────────────────┤───┘               
+           ╲│                 │╱                  
+            └─────────────────┘                                                                                     
+*/
+
+create table event_type_enm (
+    name text primary key
+        constraint only_predefined_event_types_allowed
+        check (
+            name in (
+                'every',
+                'error',
+                'audit',
+                'observation',
+                'system'
+            )
+        )
+);
+
+comment on table event_type_enm is
+'event_type_enm is an enumeration table for the valid event types within '
+'the domain';
+
+create trigger 
+  immutable_columns
+before
+update on event_type_enm
+  for each row execute procedure immutable_columns(
+      'every',
+      'error',
+      'audit',
+      'observation',
+      'system'
+      );
+
+
+create table event_format_type_enm (
+    name text primary key
+        constraint only_predefined_event_format_types_allowed
+        check (
+            name in (
+                'cloudevents-json',
+                'cloudevents-text',
+                'hclog-text',
+                'hclog-json'
+            )
+        )
+);
+
+comment on table event_format_type_enm is
+'event_format_type_enm is an enumeration table for the valid event format types '
+'within the domain';
+
+create trigger 
+  immutable_columns
+before
+update on event_format_type_enm
+  for each row execute procedure immutable_columns(
+      'cloudevents-json',
+      'cloudevents-text',
+      'hclog-text',
+      'hclog-json'
+      );
+
+create table event_config (
+    public_id wt_public_id primary key,
+    scope_id wt_scope_id not null
+        constraint iam_scope_global_fkey
+            references iam_scope_global(scope_id)
+            on delete cascade
+            on update cascade,
+        constraint scope_id_uq -- only allow one config per scope
+            unique(scope_id),
+    name wt_name,
+    constraint name_uq
```

Comment:
```suggestion
    constraint event_config_name_scope_id_uq
```
All columns in the unique constraint should be included in the constraint name